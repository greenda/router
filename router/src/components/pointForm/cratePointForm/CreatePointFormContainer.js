import React from 'react'
import PropTypes, { func, number } from 'prop-types'
import { connect } from 'react-redux'
import { addPoint } from '../../../actions/actions'
import { mapCenterSelector } from '../../../selectors/index'
import CreatePointFormView from './CreatePointFormView'

export function CreatePointFormContainer({ addPoint, mapCenter }) {
    const hundleAddPoint = (pointName) => {
        addPoint(pointName, mapCenter)
    }
    return (
        <CreatePointFormView addPoint={hundleAddPoint}/>
    )
}

CreatePointFormContainer.propTypes = {
    addPoint: func,
    mapCenter: PropTypes.shape({
        lat: number,
        lng: number,
    }),
}

export default connect(
    (store) => ({
        mapCenter: mapCenterSelector(store)
    }),
    { addPoint }
)(CreatePointFormContainer)