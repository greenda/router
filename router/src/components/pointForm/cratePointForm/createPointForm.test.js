import React from 'react';
import { CreatePointFormContainer } from './CreatePointFormContainer'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

const simulateKeyPresses = (element, characters) => {
    if (characters) {
        characters.split('').forEach(character => {
            element.simulate('keyDown', {
                which: character.charCodeAt(),
                key: character,
                keyCode: character.charCodeAt()
              });
        })
    }
}

describe('CreatePointForm', () => {
    it('render correctly point', () => {
        const element = renderer.create(
			<CreatePointFormContainer />
		).toJSON();
		expect(element).toMatchSnapshot();
    })

    it('input correct value', () => {
        const MOCK_INPUT_TEXT = 'new point name'
        const MOCK_CENTER = [100, 100]

        const hundleAddPoint = (pointName, center) => {
            expect(pointName).toEqual(MOCK_INPUT_TEXT)
            expect(center).toEqual(MOCK_CENTER)
        }
        const form = mount(
            <CreatePointFormContainer 
                mapCenter={MOCK_CENTER} 
                addPoint={hundleAddPoint} />
        )
        const inputElement = form.find('.add-point-form__input')
        
        const event = { target: { name: 'pollName', value: MOCK_INPUT_TEXT }};
        inputElement.simulate('change', event);

        inputElement.simulate('keyDown', {keyCode: 13, shiftKey: false})
    })

    it('input incorrect value (check validation)', () => {
        const MOCK_INPUT_TEXT = 'to long valu-u-u-u-u-u-u-u-u-u-ue'
        const form = mount(
            <CreatePointFormContainer />
        )
        const inputElement = form.find('.add-point-form__input')
        
        const event = { target: { name: 'pollName', value: MOCK_INPUT_TEXT }};
        inputElement.simulate('change', event)
        expect(form.find('.add-point-form__error').text())
            .toEqual('Длина названия точки - не более 30 символов')
    })
})