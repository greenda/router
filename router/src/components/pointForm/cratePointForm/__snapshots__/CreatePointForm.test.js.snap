// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePointForm render correctly point 1`] = `
<div
  className="add-point-form__container "
>
  <span
    className="add-point-form__label"
  >
    Добавить точку
  </span>
  <input
    className="add-point-form__input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Новая точка маршрута"
    type="text"
    value=""
  />
  <span
    className="add-point-form__error"
  >
    
  </span>
</div>
`;
